test/core/test_project_status.py

```diff
@@ -189,3 +189,31 @@ def test_cache_test_run_counts_on_update(self):
         self.assertEqual(2, status.test_runs_total)
         self.assertEqual(1, status.test_runs_completed)
         self.assertEqual(1, status.test_runs_incomplete)
+
+    def test_cache_regressions(self):
+        build1 = self.create_build('1', datetime=h(10))
+        test_run1 = build1.test_runs.first()
+        test_run1.tests.create(name='foo', suite=self.suite, result=True)
+        ProjectStatus.create_or_update(build1)
+
+        build2 = self.create_build('2', datetime=h(9))
+        test_run2 = build2.test_runs.first()
+        test_run2.tests.create(name='foo', suite=self.suite, result=False)
+        status = ProjectStatus.create_or_update(build2)
+
+        self.assertIsNotNone(status.regressions)
+        self.assertIsNone(status.fixes)
+
+    def test_cache_fixes(self):
+        build1 = self.create_build('1', datetime=h(10))
+        test_run1 = build1.test_runs.first()
+        test_run1.tests.create(name='foo', suite=self.suite, result=False)
+        ProjectStatus.create_or_update(build1)
+
+        build2 = self.create_build('2', datetime=h(9))
+        test_run2 = build2.test_runs.first()
+        test_run2.tests.create(name='foo', suite=self.suite, result=True)
+        status = ProjectStatus.create_or_update(build2)
+
+        self.assertIsNotNone(status.fixes)
+        self.assertIsNone(status.regressions)
```

Comment:
you could also have a test for updating fixes/regressions when new test runs arrive, to cover the update code path in `ProjectStatus.create_or_update`